src/json/json_value.cc

```diff
@@ -450,6 +450,19 @@ auto JSON::operator-=(const JSON &substractive) -> JSON & {
   }
 }
 
+[[nodiscard]] auto JSON::try_at(const JSON::String &key) const
+    -> std::optional<JSON> {
```

Comment:
@tony-go Actually, I think this interface is not great. The result will be a copy, whereas we want an optional _reference_ to `JSON`. I think you will need `std::reference_wrapper` around it though:

```suggestion
    -> std::optional<std::reference_wrapper<JSON>> {
```